blog/content/post/2022-01-03-pcie_screamer.md

```diff
@@ -111,108 +109,201 @@ To prepare `enjoy-digital/pcie_screamer`, the following steps were tried:
+```
+It turns out that `pcileech` depends on the archival version of the library
+`openssl` (which is not available as package in OS repository). The solution was
+building this library from sources. Here is sequence of command issued in console
+in order to build this library:
+```bash
+wget https://www.openssl.org/source/openssl-1.1.1o.tar.gz
+cd openssl-1.1.1o
+./config
+make
+make test
+sudo make install
+export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/lib64/:/home/ubuntu/openssl-1.1.1o/
+```
+After building the openssl library, a command was issued to test the communication
+of the PCIScreamer board with the pcileech application:
+
+```bash
+mgabryelski@maciej-HP:~/PCIScreamer$ sudo ./pcileech probe -device fpga -v
+
+DEVICE: FPGA: ERROR: Unable to load FTD3XX.dll [0,v0.0,0000]
+PCILEECH: Failed to connect to the device.
+mgabryelski@maciej-HP:~/PCIScreamer$ 
+```
+The FTDI for USB 3.0 chip is used to connect the PCIScreamer board with a PC, 
+the driver is not embedded in the Linux kernel. So I compiled the driver
+for the FTDI chip from the sources in this Github repository: [FTDI FT50x driver](https://github.com/ufrisk/pcileech/releases/tag/v4.15) The compilation was successful, 
+and after it was finished we have such files in working directory:
+
+```bash
+mgabryelski@maciej-HP:~/GIT1/pcie_screamer/drivers/ft60x$ ls
+51-ft60x.rules  ft60x.ko   ft60x.mod.c  ft60x.o     ftd3xx.h    Makefile       Module.symvers
+ft60x.c         ft60x.mod  ft60x.mod.o  ftconfig.c  loopback.c  modules.order
+mgabryelski@maciej-HP:
+```
+We try to load kernel module with FTDI driver issuing command:
+
+```bash
+mgabryelski@maciej-HP:~/PCIScreamer$ sudo insmod ft60x.ko
+````
+
+After that the kernel module is loaded:
+
+```bash
+mgabryelski@maciej-HP:~/PCIScreamer$ lsmod
+Module                  Size  Used by
+ft60x                  20480  0
+algif_skcipher         16384  0
+af_alg                 32768  1 algif_skcipher
+binfmt_misc            24576  1
+```
+Now checking `pcileech` application we have;
+```bash
+gabryelski@maciej-HP:~/PCIScreamer$ sudo ./pcileech probe -device fpga -v
+
+DEVICE: FPGA: ERROR: Unable to load FTD3XX.dll [0,v0.0,0000]
+PCILEECH: Failed to connect to the device.
+mgabryelski@maciej-HP:~/PCIScreamer$ 
+```
+As we can see, the problem still exists. I tried also building `pcileech` application 
+from sources following this tutorial [PCILeech on Linux](https://github.com/ufrisk/pcileech/releases/tag/v4.15).
+
+The following command was issued as the last test:
+```bash
+sudo ./pcileech probe -device fpga://ft2232h=1 -v
+```
+After this command such output was  seen:
+```bash
+DEVICE: FPGA: ERROR: Unable to connect to device [0,v0.0,0000]
+PCILEECH: Failed to connect to the device.
+```
+This time it wasn't information about FTDI driver, but about communication by PCI
+bus. Also firmware version hasn't been correctly read off: [0,v0.0,0000] . Unfortunately, all attempts to communicate the `pcileech` application with the 
```

Comment:
What output does pcileech give when you execute this command without having the device connected?
```
sudo ./pcileech probe -device fpga://ft2232h=1 -v
```
Is the output the same or different?

Reply:
It is the same.